cmd/po-docgen/operator.go

```diff
@@ -0,0 +1,226 @@
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+)
+
+type FlagDoc struct {
+	name         string
+	defaultValue string
+	description  string
+}
+
+func fileToASTPackage(filePaths []string) *ast.Package {
+	fset := token.NewFileSet()
+	m := make(map[string]*ast.File)
+
+	for _, filePath := range filePaths {
+
+		f, err := parser.ParseFile(fset, filePath, nil, parser.ParseComments)
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+
+		m[filePath] = f
+	}
+	apkg, _ := ast.NewPackage(fset, m, nil, nil)
+
+	return apkg
+}
+
+func unquoteLiteral(value string) string {
+	if len(value) > 0 && value[0] == '"' {
+		value = value[1:]
+	}
+
+	if len(value) > 0 && value[len(value)-1] == '"' {
+		value = value[:len(value)-1]
+	}
+	return value
+}
+
+func resolveDurationExpr(expr ast.Expr) string {
+	var exprIdent *ast.Ident
+	exprSelector, ok := expr.(*ast.SelectorExpr)
+	if ok {
+		exprIdent = exprSelector.Sel
+	} else {
+		exprIdent, ok = expr.(*ast.Ident)
+		if !ok {
+			return "No Identifier to Resolve"
+		}
+
+	}
+
+	if exprIdent.Obj == nil {
+		return exprIdent.String()
+	}
+
+	retValue := ""
+
+	if exprIdent.Obj.Kind == ast.Con {
+		valSpec := exprIdent.Obj.Decl.(*ast.ValueSpec)
+		for _, value := range valSpec.Values {
+			retValue += unquoteLiteral(value.(*ast.BasicLit).Value)
+		}
+	}
+	return retValue
+
+}
+
+func resolveBoolExpr(expr ast.Expr) string {
+	exprIdent, ok := expr.(*ast.Ident)
+	if ok {
+		return exprIdent.Name
+	}
+
+	return "No Identifier to Resolve"
+}
+
+func resolveConstStringExpr(expr ast.Expr) string {
+
+	exprBasicLit, ok := expr.(*ast.BasicLit)
+	if ok {
+		return unquoteLiteral(exprBasicLit.Value)
+	}
+
+	exprIdent, ok := expr.(*ast.Ident)
+	if ok {
+		retValue := ""
+		if exprIdent.Obj.Kind == ast.Con {
+			valSpec := exprIdent.Obj.Decl.(*ast.ValueSpec)
+			for _, value := range valSpec.Values {
+				retValue += unquoteLiteral(value.(*ast.BasicLit).Value)
+			}
+		}
+		return retValue
+	}
+
+	exprBinaryOp, ok := expr.(*ast.BinaryExpr)
+	if ok {
+		operandX := resolveConstStringExpr(exprBinaryOp.X)
+		operandY := resolveConstStringExpr(exprBinaryOp.Y)
+		if exprBinaryOp.Op == token.ADD {
+			return operandX + operandY
+		}
+		return fmt.Sprint("unhandled OP", exprBinaryOp.Op)
+	}
+
+	return ""
+}
+
+func operatorCodeToDoc(paths []string) []FlagDoc {
+	astPkg := fileToASTPackage(paths)
+	flagDocs := make([]FlagDoc, 0)
+
+	for _, file := range astPkg.Files {
+		fmt.Println(file.Name.Name)
+		for _, decl := range file.Decls {
+			funcdecl, ok := decl.(*ast.FuncDecl)
+			if ok {
+				if funcdecl.Name.Name == "init" {
```

Comment:
To avoid deeply-nested code, you can `continue` whenever a condition isn't met:

```suggestion
			funcdecl, ok := decl.(*ast.FuncDecl)
			if !ok {
				continue
			}
			
			if funcdecl.Name.Name != "init" {
				continue
			}
```

Reply:
nest level -1 :)